### PR TITLE
refactor: ibc core runtime env

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

ref: #7712 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/02-client/types/height_test.go
Comment (damiannolan) [inline modules/core/02-client/types/height_test.go:1]: revert

File: modules/core/keeper/msg_server.go
Comment (damiannolan) [inline modules/core/keeper/msg_server.go:1]: handle err returns

File: modules/core/02-client/types/height_test.go
Comment (damiannolan) [reply modules/core/02-client/types/height_test.go:1]: https://github.com/cosmos/ibc-go/pull/7601/commits/4cb4ded8b93c46e9d68693b37b0ea1689d47d18f

File: modules/core/keeper/msg_server.go
Comment (damiannolan) [reply modules/core/keeper/msg_server.go:1]: https://github.com/cosmos/ibc-go/pull/7601/commits/828f42c49e695840337f76986c692cbe87c678a8

**Comment by damiannolan** [comment]
I think I have a few follow up commits to address tomorrow, but publishing r4r. The recvPacket issue is pushed to a different PR. We could just remove the error prefixed events... but I think they were requested before for dev ex

**Review by gjermundgaraba — APPROVED** [review]
Looks straightforward enough! LGTM!

File: modules/core/02-client/abci.go [hunk 1]
```diff
@@ -9,7 +11,11 @@ import (
 )
 
 // BeginBlocker is used to perform IBC client upgrades
-func BeginBlocker(ctx sdk.Context, k *keeper.Keeper) {
+func BeginBlocker(goCtx context.Context, k *keeper.Keeper) {
+	// TODO: In order to fully migrate away from sdk.Context here we will need to depend on comet service in order
+	// to consume the full block header as Env only contains header.Info (where we cannot access next vals hash)
```
Comment (damiannolan) [inline modules/core/02-client/abci.go:16]: Create issue

File: modules/core/keeper/msg_server.go [hunk 2]
```diff
@@ -87,9 +81,7 @@ func (k *Keeper) UpgradeClient(goCtx context.Context, msg *clienttypes.MsgUpgrad
 // SubmitMisbehaviour defines a rpc handler method for MsgSubmitMisbehaviour.
 // Warning: DEPRECATED
 // This handler is redundant as `MsgUpdateClient` is now capable of handling both a Header and a Misbehaviour
-func (k *Keeper) SubmitMisbehaviour(goCtx context.Context, msg *clienttypes.MsgSubmitMisbehaviour) (*clienttypes.MsgSubmitMisbehaviourResponse, error) { //nolint:staticcheck // for now, we're using msgsubmitmisbehaviour.
-	ctx := sdk.UnwrapSDKContext(goCtx)
-
+func (k *Keeper) SubmitMisbehaviour(ctx context.Context, msg *clienttypes.MsgSubmitMisbehaviour) (*clienttypes.MsgSubmitMisbehaviourResponse, error) { //nolint:staticcheck // for now, we're using msgsubmitmisbehaviour.
```
Comment (damiannolan) [inline modules/core/keeper/msg_server.go:84]: nolint can probs be removed here

File: testing/chain.go [hunk 3]
```diff
@@ -208,8 +208,9 @@ func (chain *TestChain) GetContext() sdk.Context {
 	headerInfo := header.Info{
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply testing/chain.go:208]: Just needs dependency bump and this could be removed

**Review by damiannolan — COMMENTED** [review]
CacheContext is still being used by:
- RecvPacket
- Channel Upgrade handlers (Init, Try, Ack)

**Pull request merged**